src/neuroglancer/segmentation_display_state/base.ts

```diff
@@ -17,18 +17,41 @@
 import {SharedDisjointUint64Sets} from 'neuroglancer/shared_disjoint_sets';
 import {Uint64Set} from 'neuroglancer/uint64_set';
 import {Uint64} from 'neuroglancer/util/uint64';
+import {vec3} from 'neuroglancer/util/geom';
+import {SharedTrackableValue} from 'neuroglancer/shared_trackable_value';
+
+export interface Bounds {
+  center: vec3;
+  size: vec3;
+};
+
+export class ClipBounds extends SharedTrackableValue<Bounds> {
+  toJSON(): {}|null {
+    let bounds = this.value;
+    if (bounds === undefined) { 
+      return null;
+    }
+    let result = {
+      center: [...bounds.center.values()],
+      size: [...bounds.size.values()],
+    };
+    return result;
+  }
+}
```

Comment:
I think it would make sense to move the restore logic into a restoreState method here rather than in segmentation_user_layer.ts